packages-rs/drainpipe/src/main.rs

```diff
@@ -101,13 +98,39 @@ async fn main() {
         .await
         {
             Ok((mut socket, _response)) => {
-                println!("Connected to bgs.bsky-sandbox.dev.");
-                while let Some(Ok(Message::Binary(message))) = socket.next().await {
-                    let ctx = ctx.clone();
+                let database_url = std::env::var("DATABASE_URL").expect("DATABASE_URL not set");
+                let mut ctx = Context {
+                    frontpage_consumer_secret: std::env::var("FRONTPAGE_CONSUMER_SECRET")
+                        .expect("FRONTPAGE_CONSUMER_SECRET not set"),
+                    frontpage_consumer_url: std::env::var("FRONTPAGE_CONSUMER_URL")
+                        .expect("FRONTPAGE_CONSUMER_URL not set"),
+                    db_connection: db::db_connect(&database_url).expect("Failed to connect to db"),
+                };
+                let metrics_monitor = tokio_metrics::TaskMonitor::new();
+                {
+                    let metrics_monitor = metrics_monitor.clone();
                     tokio::spawn(async move {
-                        process(message, ctx).await;
+                        for interval in metrics_monitor.intervals() {
+                            println!("{:?} per second", interval.instrumented_count as f64 / 5.0);
+                            tokio::time::sleep(Duration::from_millis(5000)).await;
+                        }
                     });
                 }
+
+                println!("Connected to bgs.bsky-sandbox.dev.");
+                while let Some(Ok(Message::Binary(message))) = socket.next().await {
```

Comment:
The linear mentioned introducing a queue. This actually isn't needed because this while loop is already acting like a queue.

I just needed to remove `tokio::spawn` from inside (this was copy and pasted from the example) to prevent messages from being handled concurrently.